fili-core/src/test/groovy/com/yahoo/bard/webservice/util/DateTimeUtilsSpec.groovy

```diff
@@ -17,57 +23,89 @@ import spock.lang.Unroll
 class DateTimeUtilsSpec extends Specification {
 
     @Unroll
-    def "check if #c = #a + #b"() {
+    def "check if #sum = #dateTime + #timeGrain"() {
         expect:
-        DateTimeUtils.addTimeGrain(new DateTime(a), b) == new DateTime(c)
+        DateTimeUtils.addTimeGrain(new DateTime(dateTime), timeGrain) == new DateTime(sum)
 
         where:
-        a                         | b     || c
-        "2014-01-01T10:00:00.000" | HOUR  || "2014-01-01T11:00:00.000"
-        "2014-01-01"              | DAY   || "2014-01-02"
-        "2014-01-31"              | DAY   || "2014-02-01"
-        "2014-04-30"              | DAY   || "2014-05-01"
-        "2012-02-29"              | DAY   || "2012-03-01"
-        "2014-02-28"              | DAY   || "2014-03-01"
-        "2012-02-28"              | DAY   || "2012-02-29"
-        "2014-02-28"              | WEEK  || "2014-03-07"
-        "2012-02-28"              | WEEK  || "2012-03-06"
-        "2014-01-01"              | WEEK  || "2014-01-08"
-        "2014-02-01"              | WEEK  || "2014-02-08"
-        "2014-01-01"              | MONTH || "2014-02-01"
-        "2014-02-01"              | MONTH || "2014-03-01"
-        "2014-04-01"              | MONTH || "2014-05-01"
-        "2012-02-01"              | MONTH || "2012-03-01"
-        "2012-01-01"              | YEAR  || "2013-01-01"
-        "2014-07-01"              | YEAR  || "2015-07-01"
+        dateTime                  | timeGrain | sum
+        "2014-01-01T10:00:00.000" | HOUR      | "2014-01-01T11:00:00.000"
+        "2014-01-01"              | DAY       | "2014-01-02"
+        "2014-01-31"              | DAY       | "2014-02-01"
+        "2014-04-30"              | DAY       | "2014-05-01"
+        "2012-02-29"              | DAY       | "2012-03-01"
+        "2014-02-28"              | DAY       | "2014-03-01"
+        "2012-02-28"              | DAY       | "2012-02-29"
+        "2014-02-28"              | WEEK      | "2014-03-07"
+        "2012-02-28"              | WEEK      | "2012-03-06"
+        "2014-01-01"              | WEEK      | "2014-01-08"
+        "2014-02-01"              | WEEK      | "2014-02-08"
+        "2014-01-01"              | MONTH     | "2014-02-01"
+        "2014-02-01"              | MONTH     | "2014-03-01"
+        "2014-04-01"              | MONTH     | "2014-05-01"
+        "2012-02-01"              | MONTH     | "2012-03-01"
+        "2012-01-01"              | YEAR      | "2013-01-01"
+        "2014-07-01"              | YEAR      | "2015-07-01"
     }
 
-    def buildIntervalSet(List<String> intervalStrList) {
```

Comment:
This helper method has been moved to the bottom of this file for organization purpose